runtime/config.go

```diff
@@ -0,0 +1,89 @@
+package runtime
```

Comment:
My general issue with this package as described in https://github.com/pokt-network/pocket/discussions/234

Is that we essentially allow for other modules to access `module specific` configuration and genesis states.

This is not at all the intention

This seems like an error prone architecture to me

Reply:
I see what you mean but if we think about configurability in general isn't the environment (and therefore environment variables, configuration files, etc) something that's accessible everywhere within the app one way or another? 

What we should guard against is not the mere access but whenever for example inside the P2P module I use the PostgresUrl (part of persistence config) to build some logic on top of. Obviously, that's fragile and spaghetti code and bug prone because someone might just change that in persistence and consequently break P2P for example.

The environment is a `os.GetEnv()` call away and therefore any config value as well potentially.
The point is: will it make it to `main`?
Same applies for accessing configuration/genesis files.

If we want the modules to be completely isolated from each other and to bring to zero the shared code I think they should be running as separate binaries on their own isolation context (Docker container or whatnot) and communicate with the other modules via some form of well defined RPC. But we are still in monolith land here and everything is a tradeoff. Anyway, I am glad I can pick your brain about this kind of stuff. 🙏